merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -301,6 +293,7 @@ public JavaFile generateSchedulerModel(final MissionModelRecord missionModel) {
                                                           .map($ -> {
                                                             if ($.durationParameter().isPresent()) return CodeBlock.of("controllable(\"$L\")", $.durationParameter().get());
                                                             else if ($.fixedDurationExpr().isPresent()) return CodeBlock.of("fixed($L.$L)", activityTypeRecord.fullyQualifiedClass(), $.fixedDurationExpr().get());
+                                                            else if ($.parametricDuration().isPresent()) return CodeBlock.of("parametric($$ -> (new $L().new InputMapper()).instantiate($$).$L())", activityTypeRecord.inputType().mapper().name, $.parametricDuration().get());
                                                             else return CodeBlock.of("uncontrollable()");
                                                           })
                                                           .orElse(CodeBlock.of("uncontrollable()"))))
```

Comment:
Do you know why we need to set the "else uncontrollable" both in and out of the `map`?

Reply:
For some reason the effect model data is optional. If its not supplied we assume it is uncontrollable.

Reply:
It would be ideal if we could mark activities that lack an effect model with a fixed duration of zero. I don't know offhand how hard that would be to implement, and I don't expect that to be done in this PR

Reply:
it would be pretty trivial

Reply:
#911 never forget